File: ios/IdentityKeyPair.swift
import Foundation
import ExpoModulesCore
import LibSignalClient

final class IdentityKeyPairRef: SharedObject {
  let keyPair: IdentityKeyPair

  init(keyPair: IdentityKeyPair) {
    self.keyPair = keyPair
    super.init()
  }
}

final class PublicIdentityKeyRef: SharedObject {
  let key: IdentityKey

  init(key: IdentityKey) {
    self.key = key
    super.init()
  }
}

final class PrivateKeyRef: SharedObject {
  let key: PrivateKey

  init(key: PrivateKey) {
    self.key = key
    super.init()
  }
}

File: android/src/main/java/expo/modules/libsignal/IdentityKeyPair.kt
package expo.modules.libsignal

import expo.modules.kotlin.sharedobjects.SharedObject
import org.signal.libsignal.protocol.IdentityKey
import org.signal.libsignal.protocol.IdentityKeyPair as SignalIdentityKeyPair
import org.signal.libsignal.protocol.ecc.ECPrivateKey

class IdentityKeyPairRef(val keyPair: SignalIdentityKeyPair) : SharedObject()

class PublicIdentityKeyRef(val key: IdentityKey) : SharedObject()

class PrivateKeyRef(val key: ECPrivateKey) : SharedObject()
